### ISSUE TITLE
Issue with DCSource in Brian when value changing values on the fly

### ISSUE THREAD

**Issue by appukuttan-shailesh** [NeuralEnsemble/PyNN#451]
The code [here](https://github.com/appukuttan-shailesh/PyNN_testData/blob/master/Test_DC_change_fly.py) attempts to change the values of the DCSource on the fly. The amplitude is 0 for the first 100 ms, and thereafter changed for t > 100 ms. But as the start and stop times are specified less than 100 ms, there should be no effect on the membrane potential. As seen below the output in Brian shows an erroneous value at the transition.

Brian output (incorrect):
![brian_bad](https://cloud.githubusercontent.com/assets/24866517/23460240/aaed2a72-fe84-11e6-8ea6-806ed4c397c3.png)
NEURON output (correct):
![neuron_good](https://cloud.githubusercontent.com/assets/24866517/23460251/b5ae10de-fe84-11e6-8c18-4601c3b2f576.png)

**Comment by appukuttan-shailesh** [comment]
This can be rectified by incorporating the below mentioned changes in https://github.com/NeuralEnsemble/PyNN/blob/master/pyNN/brian/standardmodels/electrodes.py.
There are two main changes:
1) the index (_self.i_) to the time and amplitudes list is reset to zero only at the start of a new run, and not for continuation of existing runs (which can involve changes on the fly). This involves changing:
``` python
    def _reset(self):
        self.i = 0
        self.running = True
        if self._is_computed:
            self._generate()
```
to
``` python
    def _reset(self):
        if not hasattr(self, 'running') or self.running == False:
            self.i = 0
            self.running = True
        if self._is_computed:
            self._generate()
```

2) Adding the current simulation time and corresponding amplitude in the times (_self.times_) and amplitudes (_self.amplitudes_) list, while ensuring that the temporal ordering is maintained. This is achieved by modifying the _generate() method for DCSource from:
``` python
    def _generate(self):
        if self.start == 0:
            self.times = [self.start, self.stop]
            self.amplitudes = [self.amplitude, 0.0]
        else:
            self.times = [0.0, self.start, self.stop]
            self.amplitudes = [0.0, self.amplitude, 0.0]
```
to
``` python
    def _generate(self):
        if self.start == 0:
            self.times = [self.start, self.stop]
            self.amplitudes = [self.amplitude, 0.0]
        else:
            self.times = [0.0, self.start, self.stop]
            self.amplitudes = [0.0, self.amplitude, 0.0]
        if self.start < simulator.state.t*1e-3 < self.stop:
            self.times.insert(-1, simulator.state.t*1e-3)
            self.amplitudes.insert(-1, self.amplitude)
            if (self.start==0 and self.i==2) or (self.start!=0 and self.i==3):
                self.i -= 1
```

These changes were tested against various start/stop combos for the DCSource described in this [file](https://github.com/appukuttan-shailesh/PyNN_testData/blob/master/Test_DC_change_fly_v2.py). The earlier test [file](https://github.com/appukuttan-shailesh/PyNN_testData/blob/master/Test_DC_change_fly.py) was not used owing to problems arising from issue #452 .

_Note: This seems to resolve the issue of changing parameters on the fly for the DCSource. It would be useful to have this checked for the other electrode classes._

**Comment by apdavison** [comment]
@appukuttan-shailesh thanks for investigating this. Can you create a pull request with these changes (and a test), please?

**Comment by apdavison** [comment]
Fixed in #467
